pkg/controller/job/controller.go

```diff
@@ -255,6 +258,16 @@ func (jm *JobController) enqueueController(obj interface{}) {
 	// all controllers there will still be some replica instability. One way to handle this is
 	// by querying the store for all controllers that this rc overlaps, as well as all
 	// controllers that overlap this rc, and sorting them.
+	// Instead of insert the key directly, check if this key is in the queue already
+	// If it isn't or this key is in the head of the queue, insert the key, otherwise, abort.
+	// If the controller fail and restart, both queue and queuedJob will be lost
+	// everything need to be start over.
+	if jm.queuedJob.Has(key) {
```

Comment:
it needs double check if the key is in the head. If the key is in the head, it may in any stage of sync process. In this situation, this update may not be synchronized in that sync process(for example, the sync process already come to the end but it has not remove this key yet).